File: modules/widget-bridge/ios/WidgetBridgeModule.swift
import ExpoModulesCore
import WidgetKit

public class WidgetBridgeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("WidgetBridge")

    Function("setSharedData") { (appGroupId: String, key: String, json: String) in
      UserDefaults(suiteName: appGroupId)?.set(json, forKey: key)
    }

    Function("reloadWidgets") { () in
      WidgetCenter.shared.reloadAllTimelines()
    }
  }
}

File: modules/widget-bridge/src/WidgetBridgeModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class WidgetBridgeModule extends NativeModule<{}> {
  setSharedData(appGroupId: string, key: string, json: string): void;
  reloadWidgets(): void;
}

export default requireNativeModule<WidgetBridgeModule>('WidgetBridge');
